servers/rendering/rendering_device.cpp

```diff
@@ -242,8 +242,16 @@ RID RenderingDevice::_uniform_set_create(const Array &p_uniforms, RID p_shader,
 	return uniform_set_create(uniforms, p_shader, p_shader_set);
 }
 
-Error RenderingDevice::_buffer_update(RID p_buffer, uint32_t p_offset, uint32_t p_size, const Vector<uint8_t> &p_data, uint32_t p_post_barrier) {
-	return buffer_update(p_buffer, p_offset, p_size, p_data.ptr(), p_post_barrier);
+template <typename T>
+Error RenderingDevice::_buffer_update_array(RID p_buffer, uint32_t p_index, uint32_t p_size, const Vector<T> &p_data, uint32_t p_post_barrier) {
+	size_t size = sizeof(T);
+	return buffer_update(p_buffer, p_index * size, p_size * size, (uint8_t *)p_data.ptr(), p_post_barrier);
+}
+
```

Comment:
As I mentioned, updating an element should be discouraged, its not even possible with many of the buffer types due to granularity.